### PR TITLE
refactor(apps/price_pusher): crash on RPC failures

### PR DESCRIPTION
**Pull request by ali-bahjati**
This PR updates the price pusher code to cause a crash when the RPC has issues. The RPC issues can be either the target network or Hermes.

This also solves #1704.

### CODE REVIEW

File: apps/price_pusher/src/sui/sui.ts [hunk 1]
```diff
@@ -383,25 +363,20 @@ export class SuiPricePusher implements IPricePusher {
...
-      return ref;
+    const objectResponse = await provider.getObject({ id: ref.objectId });
+    if (objectResponse.data !== undefined) {
+      return {
+        digest: objectResponse.data!.digest,
+        objectId: objectResponse.data!.objectId,
+        version: objectResponse.data!.version,
+      };
+    } else {
+      throw new Error("Failed to refresh object reference");
```
Comment (m30m) [inline apps/price_pusher/src/sui/sui.ts:379]: I'm not sure how often this may happen, but if it fails frequently and causes crashes, there is a higher chance that we lock all the gas price objects

Comment (ali-bahjati) [reply apps/price_pusher/src/sui/sui.ts:379]: During my experiments I think I've resolved the issue of locked gases. It seems that it happens when we reuse a gas coin upon crash/restarts. I changed our deployment configuration to never allow to live instances at the same time and wait 10s before starting a new instance to make sure older ones are settled. No locking issues since this change.

I'll also run this code for sui for a day or so before merging to see how it works.

File: apps/price_pusher/src/solana/solana.ts [hunk 2]
```diff
@@ -28,6 +28,25 @@ export class SolanaPriceListener extends ChainPriceListener {
     super(config.pollingFrequency, priceItems);
   }
 
+  // Checking the health of the Solana connection by checking the last block time
+  // and ensuring it is not older than 30 seconds.
+  private async checkHealth() {
```
Comment (guibescos) [inline apps/price_pusher/src/solana/solana.ts:33]: should we have a health check for Jito?

Comment (ali-bahjati) [reply apps/price_pusher/src/solana/solana.ts:33]: i throw an error on the failures there. Didn't do on Solana because it might fail due to landing issues.

File: apps/price_pusher/src/solana/solana.ts [hunk 3]
```diff
@@ -28,6 +28,25 @@ export class SolanaPriceListener extends ChainPriceListener {
...
+    const slot = await this.pythSolanaReceiver.connection.getSlot();
+    const blockTime = await this.pythSolanaReceiver.connection.getBlockTime(
+      slot
+    );
+    if (blockTime !== undefined || blockTime < Date.now() / 1000 - 30) {
```
Comment (guibescos) [inline apps/price_pusher/src/solana/solana.ts:38]: this boolean is wrong
`if (blockTime && blockTime < Date.now() / 1000 - 30)`

File: apps/price_pusher/src/solana/solana.ts [hunk 4]
```diff
@@ -28,6 +28,25 @@ export class SolanaPriceListener extends ChainPriceListener {
...
+      throw new Error("Solana connection is unhealthy");
+    }
+  }
+
+  async start() {
+    // Frequently check the RPC connection to ensure it is healthy
+    setInterval(() => this.checkHealth.bind(this), 5000);
```
Comment (guibescos) [inline apps/price_pusher/src/solana/solana.ts:45]: this needs to be `setInterval(this.checkHealth.bind(this), 5000);`

**Review by guibescos — CHANGES_REQUESTED** [review]
I debugged it a bit, found couple of bugs

File: apps/price_pusher/src/sui/sui.ts [hunk 5]
```diff
@@ -337,7 +317,7 @@ export class SuiPricePusher implements IPricePusher {
     ignoreGasObjects: string[],
     logger: Logger
   ): Promise<SuiObjectRef[]> {
-    const signerAddress = await signer.toSuiAddress();
+    const signerAddress = signer.toSuiAddress();
```
Comment (ali-bahjati) [inline apps/price_pusher/src/sui/sui.ts:320]: drive-by: linter

File: apps/price_pusher/src/sui/sui.ts [hunk 6]
```diff
@@ -208,12 +194,6 @@ export class SuiPricePusher implements IPricePusher {
       provider,
       logger,
       priceServiceConnection,
-      pythPackageId,
-      pythStateId,
-      wormholePackageId,
-      wormholeStateId,
-      endpoint,
-      keypair,
```
Comment (ali-bahjati) [inline apps/price_pusher/src/sui/sui.ts:216]: drive-by: linter

File: apps/price_pusher/src/solana/solana.ts [hunk 7]
```diff
@@ -125,6 +125,7 @@ export class SolanaPricePusher implements IPricePusher {
       this.logger.info({ signatures }, "updatePriceFeed successful");
     } catch (err: any) {
       this.logger.error(err, "updatePriceFeed failed");
+      return;
```
Comment (guibescos) [inline apps/price_pusher/src/solana/solana.ts:128]: this return doesn't seem to do anything

**Review by guibescos — APPROVED** [review]

File: apps/price_pusher/src/solana/solana.ts (hunk 7 above)
Comment (ali-bahjati) [reply apps/price_pusher/src/solana/solana.ts:128]: lol it was there before and i removed it. added it back to minimize the changes.
